release-notes/7.0/7.0.0/api-diff/Microsoft.NETCore.App/7.0.0_System.IO.md

```diff
@@ -0,0 +1,156 @@
++        [UnsupportedOSPlatformAttribute("windows")]
++        public static void SetUnixFileMode(string path, UnixFileMode mode);
+     }
+     public sealed class FileInfo : FileSystemInfo {
+-        public override string ToString();
+     }
+     public class FileStream : Stream {
+-        [UnsupportedOSPlatformAttribute("ios")]
+-        [UnsupportedOSPlatformAttribute("macos")]
+-        [UnsupportedOSPlatformAttribute("tvos")]
+-        public virtual void Lock(long position, long length);
++        [UnsupportedOSPlatformAttribute("freebsd")]
++        [UnsupportedOSPlatformAttribute("ios")]
++        [UnsupportedOSPlatformAttribute("macos")]
++        [UnsupportedOSPlatformAttribute("tvos")]
++        public virtual void Lock(long position, long length);
+-        [UnsupportedOSPlatformAttribute("ios")]
+-        [UnsupportedOSPlatformAttribute("macos")]
+-        [UnsupportedOSPlatformAttribute("tvos")]
+-        public virtual void Unlock(long position, long length);
++        [UnsupportedOSPlatformAttribute("freebsd")]
++        [UnsupportedOSPlatformAttribute("ios")]
++        [UnsupportedOSPlatformAttribute("macos")]
++        [UnsupportedOSPlatformAttribute("tvos")]
++        public virtual void Unlock(long position, long length);
+     }
+     public sealed class FileStreamOptions {
++        public UnixFileMode? UnixCreateMode { get; [UnsupportedOSPlatformAttribute("windows")] set; }
+     }
+     public static class FileSystemAclExtensions {
+-        public static FileStream Create(this FileInfo fileInfo, FileMode mode, FileSystemRights rights, FileShare share, int bufferSize, FileOptions options, FileSecurity fileSecurity);
++        public static FileStream Create(this FileInfo fileInfo, FileMode mode, FileSystemRights rights, FileShare share, int bufferSize, FileOptions options, FileSecurity? fileSecurity);
+     }
+     public abstract class FileSystemInfo : MarshalByRefObject, ISerializable {
++        public UnixFileMode UnixFileMode { get; [UnsupportedOSPlatformAttribute("windows")] set; }
+     }
+     public class FileSystemWatcher : Component, ISupportInitialize {
++        public WaitForChangedResult WaitForChanged(WatcherChangeTypes changeType, TimeSpan timeout);
+     }
+     public static class Path {
++        public static bool Exists([NotNullWhenAttribute(true)] string? path);
+     }
+     public static class RandomAccess {
++        public static void SetLength(SafeFileHandle handle, long length);
+     }
+     public abstract class Stream : MarshalByRefObject, IAsyncDisposable, IDisposable {
+-        [ObsoleteAttribute("Stream.CreateWaitHandle has been deprecated. Use the ManualResetEvent(false) constructor instead.")]
+-        protected virtual WaitHandle CreateWaitHandle();
++        [ObsoleteAttribute("CreateWaitHandle has been deprecated. Use the ManualResetEvent(false) constructor instead.")]
++        protected virtual WaitHandle CreateWaitHandle();
```

Comment:
Only the text for `Obsolete` has changed, I don't believe it's worth including in API diff.

```suggestion
```